client/connector/FileWatcher.cs

```diff
@@ -1,23 +1,24 @@
 ﻿using System;
 using System.IO;
 using io.harness.cfsdk.client.api;
-using Serilog;
+using Microsoft.Extensions.Logging;
 
 namespace io.harness.cfsdk.client.connector
 {
-    public class FileWatcher : IDisposable, IService
+    internal sealed class FileWatcher : IService
```

Comment:
Note: this public class is now internal. Since it is internal, we can just call the `Stop()` method instead of `Dispose()`.